tests/providers/aws/aws_e2e.py

```diff
@@ -132,22 +150,35 @@ def test_end_to_end_vm(self):
     instance = self.aws.ResourceApi(EC2_SERVICE).Instance(
         self.analysis_vm.instance_id)
     self.assertEqual(instance.instance_id, self.analysis_vm.instance_id)
-    self.assertIn(volume_to_attach.volume_id,
+    self.assertIn(volume_copy.volume_id,
                   [vol.volume_id for vol in instance.volumes.all()])
 
+  def VerifyVolume(self, aws_account, volume_id):
+    """Util method for AWS.
+
+    Verify that a volume created by the library is in AWS.
+
+    Args:
+      aws_account (AWSAccount): The AWS account to use.
+      volume_id (str): The volume ID of the volume to verify.
+    """
+    self.volumes.append(
+        (aws_account, aws_account.ResourceApi(EC2_SERVICE).Volume(volume_id)))
+    self.assertEqual(self.volumes[-1][1].volume_id, volume_id)
+
```

Comment:
I think tests would be clearer if the actual assertion was in the tests, rather than hidden in a function. (It doesn't matter if you repeat yourself in tests - it's better they are explicit than very elegant and convoluted)

I see you're saving modules for cleanup at the end. In that case, it might be better to have a function very explicitly named something like `_StoreVolumeForTestCleanup`. that does the appending. Anyone reading the test will know what it's about and that it's probably safe to ignore.

Reply:
Thanks for the clarification, changed :)